k9mail-library/src/main/java/com/fsck/k9/mail/ssl/TrustedSocketFactory.java

```diff
@@ -10,4 +11,6 @@
 public interface TrustedSocketFactory {
     Socket createSocket(Socket socket, String host, int port, String clientCertificateAlias)
             throws NoSuchAlgorithmException, KeyManagementException, MessagingException, IOException;
+
+    ProxySettings getProxySettings();
```

Comment:
This doesn't belong in `TrustedSocketFactory`.

We probably want a general socket factory that knows how to create trusted and regular sockets. Then all the proxy code can be moved there.